File: Python/344. Reverse String.py
"""
Write a function that takes a string as input and returns the string reversed.

Example:Given s = "hello", return "olleh".
"""


class Solution:
    def reverseString(self, s):
        """
        :type s: str
        :rtype: str
        """
        res = str()
        for item in s:
            res = item + res

        return res

if __name__ == '__main__':
    solution = Solution()
    print(solution.reverseString("hello"))
    print(1)

File: C++/344. Reverse String.cpp
//
// Created by 王阳 on 2018/3/27.
//

/*
 * Write a function that takes a string as input and
 * returns the string reversed.
 * Example:
 * Given s = "hello", return "olleh".
 */

#include <iostream>
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    string reverseString(string s) {
        int length = int(s.length());
        int i = 0;
        int j = length - 1;
        while (i < j) {
            char tmp = s[i];
            s[i] = s[j];
            s[j] = tmp;
            i++;
            j--;
        }
        return s;
    }
};

int main() {
    Solution *solution = new Solution();
    string s("hello");
    printf("%s\n", solution->reverseString(s));
    return 0;
}